src/ring_theory/localization.lean

```diff
@@ -2373,6 +2398,42 @@ begin
     rw [← h.symm.map_mul, hz, h.symm.map_zero] }
 end
 
+variable (M)
+
+lemma is_fraction_ring_of_is_localization (S T : Type*) [comm_ring S] [comm_ring T]
+  [algebra R S] [algebra R T] [algebra S T] [is_scalar_tower R S T]
+  [is_localization M S] [is_fraction_ring R T] (hM : M ≤ non_zero_divisors R) :
+    is_fraction_ring S T :=
+begin
+  have := is_localization_of_submonoid_le S T M (non_zero_divisors R) _,
+  refine @@is_localization_of_is_exists_mul_mem _ _ _ _ _ _ this _ _,
+  { exact map_non_zero_divisors_le M S },
+  { rintro ⟨x, hx⟩,
+    obtain ⟨⟨y, s⟩, e⟩ := is_localization.surj M x,
+    use algebra_map R S s,
+    rw [mul_comm, subtype.coe_mk, e],
+    refine set.mem_image_of_mem (algebra_map R S) _,
+    intros z hz,
+    apply is_localization.injective S hM,
+    rw map_zero,
+    apply hx,
+    rw [← (map_units S s).mul_left_inj, mul_assoc, e, ← map_mul, hz, map_zero, zero_mul] },
+  { exact hM }
+end
+
+lemma is_fraction_ring_of_is_domain_of_is_localization [is_domain R] (S T : Type*)
+  [comm_ring S] [nontrivial S] [comm_ring T] [algebra R S] [algebra R T] [algebra S T]
```

Comment:
I think you shouldn't need `nontrivial S`. You don't need it for the statement, right? And `S` maps to `T`, which is the fraction ring of `R`. So `T` is nontrivial, hence `S` is nontrivial. If the statement doesn't need the assumption, then I would leave it out.